code/game/objects/items/devices/transfer_valve.dm

```diff
@@ -133,24 +133,21 @@
 		. += "proxy_beam"
 
 
-/obj/item/transfer_valve/proc/merge_gases(datum/gas_mixture/target, change_volume = TRUE)
-	var/target_self = FALSE
-	var/datum/gas_mixture/mix_one = tank_one.return_air()
-	var/datum/gas_mixture/mix_two = tank_two.return_air()
-	if(!target || (target == mix_one))
-		target = mix_two
-	if(target == mix_two)
-		target_self = TRUE
+/// Merge both gases into a single tank. Combine the volume by default. If target tank isn't specified default to tank_two
+/obj/item/transfer_valve/proc/merge_gases(obj/item/tank/target = tank_two, change_volume = TRUE)
+	if(!istype(target) && target != tank_one && target != tank_two)
+		return FALSE
```

Comment:
Defaulting to other vars in an argument def is really kinda hard to parse code, it'd be better if you moved it into an explicit if block , somewhat like the old code